### PR TITLE
Alerts refactor

### LINKED ISSUE #229: Remove DuffekDialog
I don't recall the reasoning for using DuffekDialog, but here are the reasons we should be using the system alert:

- There's a bug where they can't be dismissed under certain conditions: https://github.com/AmericanWhitewater/aw-ios/pull/228.
- The system alerts work fine to my knowledge, and will be less code.
- The system alerts adhere to Apple styling.

### PR DESCRIPTION
**Pull request by yeahphil**
This PR removes or refactors most alerts in the app. 

It doesn't *quite* remove DuffekDialog or its external dependency on NYAlertController, leaving a couple uses: a date picker and the runnable range picker, used when reporting a flow or posting a photo. 

In several places, an alert was used to deal with an error state the app just shouldn't get into, and I've removed the alert and either refactored to make the state impossible or done some light UI adustment to prevent it. Notes in the commit messages for these.

This PR also refactors and removes the Location singleton, turning it into extensions on CLLocationManager and MKMapView, and a static helper that returns a UIAlertController with a 'location denied' message. When that alert is shown, a view controller is now always responsible for presenting it.

Fixes #229, more or less

- Remove funcs with no callers from DuffekDialog
- Replace straightforward OK dialogs with UIAlertController
- Remove SignInViewController.referenceViewController, which had no callers
- Change to a little more ergonomic way of making SignInViewControllers
- Remove dialog presentation from within AWGQLApiHelper
- Remove all but one use of DuffekDialog.showStandardDialog
- Remove empty submit alert from AddAlertTableViewController
- Remove an error alert that would never be shown
- Remove an alert presented if AddRiverFlowTableViewController had no selected run
- Remove error alerts from GalleryViewController
- Remove no longer used parts of DuffekDialog
- Unify the 2 Location.checkLocationStatus funcs, adding a param
- Turn Location.hasLocation(mapView:) into MKMapView.hasLocation
- Remove Location singleton
- Remove DuffekDialog.showStandardDialog, now unused
- Convert DuffekDialog.showPickerDialog -> pickerDialog, showDatePickerDialog -> datePickerDialog
- Remove DuffekDialog's instance state and displayAlert(), no longer used.

### CODE REVIEW

File: American Whitewater/ViewControllers/More/MoreTableViewController.swift [hunk 1]
```diff
@@ -65,17 +61,26 @@ class MoreTableViewController: UITableViewController, MFMailComposeViewControlle
         } else if indexPath.row == 3 {
             
             // FEEDBACK - via email
+    
+            // AWTODO: is this still the right email to use?
+            let toAddress = "evan@americanwhitewater.org"
```
Comment (Gregliest) [inline American Whitewater/ViewControllers/More/MoreTableViewController.swift:66]: Yes, for now. Although maybe it should be pulled out into a constant?

File: American Whitewater/ViewControllers/Reaches/RiverFlowControllers/AddRiverFlowTableViewController.swift [hunk 2]
```diff
@@ -25,19 +26,57 @@ class AddRiverFlowTableViewController: UITableViewController {
...
+    }
     
     var unitOptions = ["cfs", "ft", "in"]
     var availableMetrics = [String:String]()
     
     let dateFormatter = DateFormatter()
     var awImagePicker: AWImagePicker!
     
-    var flowObservationValue: Double? = nil
+    // AWTODO: This is the state for the storyboard placeholder that was being shown by default. Is it the correct default?
```
Comment (Gregliest) [inline American Whitewater/ViewControllers/Reaches/RiverFlowControllers/AddRiverFlowTableViewController.swift:60]: Hmmm, maybe it should show a nil state to make sure the user enters something here? We don't want things defaulting to runnable if they aren't.

**Review by Gregliest — APPROVED** [review]
Generally looks great! I didn't go super in depth in the logic, especially in onboarding. Lmk if there's something in particular I should double check. Otherwise, looks good!

File: American Whitewater/ViewControllers/More/MoreTableViewController.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (yeahphil) [reply American Whitewater/ViewControllers/More/MoreTableViewController.swift]: Cool -- extracted

File: American Whitewater/ViewControllers/Reaches/RiverFlowControllers/AddRiverFlowTableViewController.swift (hunk 2 above)
Comment (yeahphil) [reply American Whitewater/ViewControllers/Reaches/RiverFlowControllers/AddRiverFlowTableViewController.swift:60]: Makes sense -- done

**Pull request merged**